mapr/datadog_checks/mapr/mapr.py

```diff
@@ -50,28 +54,38 @@ def __init__(self, name, init_config, instances):
         )
         self.allowed_metrics = [re.compile(w) for w in self.instance.get('metric_whitelist', [])]
         self.base_tags = self.instance.get('tags', [])
-        self.is_first_check_run = True
+        self.has_ever_submitted_metrics = False
+
+        auth_ticket = self.instance.get('ticket_location', os.environ.get(TICKET_LOCATION_ENV_VAR))
 
-        auth_ticket = self.instance.get('ticket_location')
-        if auth_ticket:
-            os.environ['MAPR_TICKETFILE_LOCATION'] = auth_ticket
-        elif not os.environ.get('MAPR_TICKETFILE_LOCATION'):
+        if not auth_ticket:
             self.log.warning(
-                "MAPR_TICKETFILE_LOCATION environment variable not set, this may cause authentication issues"
+                "Neither `ticket_location` (in the config.yaml) or the %s environment variable is set. This will"
+                "cause authentication issues if your cluster requires authenticated requests.",
+                TICKET_LOCATION_ENV_VAR,
+            )
+        elif not os.access(auth_ticket, os.R_OK):
+            raise CheckException(
+                "MapR authentication ticket located at %s is not readable by the dd-agent "
+                "user. Please update the file permissions.",
+                auth_ticket,
             )
 
+        os.environ[TICKET_LOCATION_ENV_VAR] = auth_ticket
+
     def check(self, _):
         if ck is None:
             raise CheckException(
                 "confluent_kafka was not imported correctly, make sure the library is installed and that you've "
-                "set LD_LIBRARY_PATH correctly. Please refer to datadog documentation for more details."
+                "set LD_LIBRARY_PATH correctly. Please refer to datadog documentation for more details. Error is %s"
+                % ck_import_error
             )
 
         try:
             conn = self.get_connection()
         except Exception:
             self.service_check(
-                SERVICE_CHECK, AgentCheck.CRITICAL, self.base_tags + ['topic_path:{}'.format(self.topic_path)]
+                SERVICE_CHECK, AgentCheck.CRITICAL, self.base_tags + ['topic:{}'.format(self.topic_path)]
```

Comment:
why this change?

Reply:
That was a minor mistake not caught earlier, the `OK` service check is sent with `topic`.

Reply:
Can we put this in a separate PR? or put in title

Reply:
Done